src/key-value-engine/introduction.md

```diff
@@ -0,0 +1,29 @@
+# Key-Value Engine
+
+A key-value engine serves at the bottommost layer in a key-value
+database, unless you are going to build your own file system or
+operating system. A key-value engine is crucial for a database because
+it manages all the persistent data directly.
+
+Most key-value engines provide some common interfaces like `Get`,
+`Put` and `Delete`. Some engines also allow you to iterate the
+key-values in order efficiently. Usually, a storage engine provides
+some special features more or less to let you take good use of it too.
+
+Choosing a key value engine is the first step to build a
+database. Here are some important things we need to consider:
+
+- The data storage model. Different models are optimized for different
```

Comment:
I use data structure instead of data storage model in `B-Tree vs Log-Stuctured Merge-Tree`. I think we need to be consistent.